pkg/operator/ceph/cluster/mon/health.go

```diff
@@ -55,6 +55,9 @@ func NewHealthChecker(monCluster *Cluster) *HealthChecker {
 
 	monCRDTimeoutSetting := monCluster.spec.HealthCheck.DaemonHealth.Monitor.Timeout
 	if monCRDTimeoutSetting != "" {
+		if monCRDTimeoutSetting == "0" {
```

Comment:
How about checking for 0 after it's parsed? Then if they have "0s" it will still evaluate to 0.

Reply:
It's possible but what's the benefit over the current approach?

Reply:
What happens if the user sets `0s` or `0m` in the rest of the code? Does it still effectively disable health check timeout?

Reply:
No, it has to be 0 so 0s or 0m won't work. I don't think this is realistic but we can check for this after the time parsing.